### PR TITLE
stop logging `cloud_init_bytes` in sled-agent

### LINKED ISSUE #6387: Sled-agent could demote printing cloud init bytes in the instance-ensure request to debug level
I could have sworn there was already an issue for this, but I can't find it. We seem to still be printing the cloud-init bytes in the instance-ensure log messages in the sled agent. That could probably be demoted to debug level at this point, since it's quite big and we have not needed to inspect the contents for quite some time.

### PR DESCRIPTION
**Pull request by iliana**
aka "want sled-agent to stop screaming at base64 floppy disks"

Fixes #6387.

1. Create a `NoDebug` wrapper struct for this purpose in omicron-common. (Note that the OpenAPI schema for sled-agent does not change; schemars seems to honor `#[serde(transparent)]`.)
2. Apply `NoDebug` inside the `cloud_init_bytes` `Option`. Thus we log everything else about the instance hardware, and whether or not there is cidata, but not the cidata itself.
3. Demote an INFO-level log of the Propolis client's `InstanceEnsureRequest` to DEBUG.

### CODE REVIEW

File: common/src/lib.rs [hunk 1]
```diff
@@ -118,3 +118,22 @@ where
 async fn never_bail() -> Result<bool, Error> {
     Ok(false)
 }
+
+/// A wrapper struct that does nothing other than elide the inner value from
```
Comment (sunshowers) [inline common/src/lib.rs:122]: I wrote https://docs.rs/debug-ignore a bit ago which we use :)

**Review by hawkw — APPROVED** [review]
Thank you for doing this, the giant base64 strings were getting on my nerves a bit!

File: common/src/lib.rs (hunk 1 above)
Comment (iliana) [reply common/src/lib.rs:122]: The trick is that it must also implement `JsonSchema` for this particular use case, and anything else we ever want to use it in a struct for as well. For that reason it seems to make sense to keep it within Omicron so we can adjust it as needed.

Comment (sunshowers) [reply common/src/lib.rs:122]: sighhhh, yeah.

File: sled-agent/types/src/instance.rs [hunk 2]
```diff
@@ -63,7 +64,7 @@ pub struct InstanceHardware {
     pub dhcp_config: DhcpConfig,
     // TODO: replace `propolis_client::*` with locally-modeled request type
     pub disks: Vec<propolis_client::types::DiskRequest>,
-    pub cloud_init_bytes: Option<String>,
+    pub cloud_init_bytes: Option<NoDebug<String>>,
```
[the remark this answers is not in the document]
Comment (sunshowers) [reply sled-agent/types/src/instance.rs:67]: FWIW I prefer the wrapper struct over the two other alternatives I'm aware of, the manual impl you mentioned and the derive-more crate.

File: common/src/lib.rs (hunk 1 above)
Comment (sunshowers) [reply common/src/lib.rs:122]: (Could you add a comment about this?)

File: sled-agent/types/src/instance.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (hawkw) [reply sled-agent/types/src/instance.rs:64]: For the record, the wrapper struct is fine with me!

**Pull request merged**